server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/java/org/apache/james/jmap/cassandra/CassandraSetMessagesMethodTest.java

```diff
@@ -68,5 +90,89 @@ protected MessageId randomMessageId() {
     public void attachmentsShouldBeRetrievedWhenChainingSetMessagesAndGetMessagesTextAttachment() throws Exception {
 
     }
-    
+
+    @Test
+    public void setMessagesShouldWorkForHugeNumberOfEmailsToTrashWhenChunksConfigurationAreLowEnough() throws Throwable {
+        teardown();
+        cassandra = new DockerCassandraRule(configuration -> configuration.addProperty("chunk.size.expunge", 85));
+        setup();
```

Comment:
Can't you use this option for the entire class? Or maybe if you really want to make it clear that this option is related to that test, then please extract it somewhere else...

Reply:
We've decided to have a test with a "failing" configuration, so it's test-based.

I struggle to find the right way to do it, where do you think I can extract it? Thanks in advance.

Reply:
Let's create a new test class `CassandraJMAPBulkOperationTest`

That way you can set bulk properties there without affecting other tests.

Reply:
I still don't see how to have a per-test parameterization, have you some hints about that? Thanks

Reply:
It's not per test. It's put the tests that need the 86 expunge chunk size in the same class.

Reply:
I added two tests: one working (with a chunk size of 85) and one failing (with a chunk size of 250).
If I understand correctly, I should write two classes? It seems exaggerated.

Reply:
Or in the additionnal class, start James **in** the test method (not in setUp())....

Reply:
But don't call setUp(), teardown() from within the test method...